### PR TITLE
Add conda-solver option

### PR DESCRIPTION
**Pull request by jaimergp**
Github Actions runners now ship Miniconda 23.5.2, which bundles conda-libmamba-solver (although it's not enabled by default). `conda` will switch to `libmamba` as the new default solver soon, so we better be prepared and provide configurability here to (A) opt-in already, (B) opt-out if it doesn't work.

### CODE REVIEW

File: src/base-tools/update-mamba.ts [hunk 1]
```diff
@@ -14,9 +14,6 @@ export const updateMamba: types.IToolProvider = {
   provides: async (inputs, options) =>
     inputs.mambaVersion !== "" || options.mambaInInstaller,
   toolPackages: async (inputs, options) => {
-    core.warning(
```
Comment (goanpeca) [inline src/base-tools/update-mamba.ts:17]: So mamba support is no longer experimental?

Should we still warn about "differently solved environments!" @jaimergp ?

**Review by goanpeca — COMMENTED** [review]
@jaimergp just to double check, are the options being automatically used with the solver, or do we need to update the code to actually include the use of such option? 🤔

**Comment by jaimergp** [comment]
`conda` will take the solver from `.condarc`, so I think that's all we need. You can [see in the logs](https://github.com/conda-incubator/setup-miniconda/actions/runs/6199167070/job/16831134189?pr=291#step:3:267) how it's already in use (that kind of channels, platform output is libmamba only).

**Comment by tobiasdiez** [comment]
Any update on this one here? Would be nice to be able to use the new conda solver!

**Comment by jaimergp** [comment]
Hi @tobiasdiez, in the meantime, you can use `CONDA_SOLVER` (env var). It will only apply to the jobs you are using it on, though (it's not a persistent configuration like `conda-solver`), but should be enough to get you started:

```yaml
      - uses: conda-incubator/setup-miniconda@v2
        env:
          CONDA_SOLVER: libmamba
        with:
          auto-update-conda: true
          python-version: "3.9"
```

**Comment by goanpeca** [comment]
@jaimergp is the failing test related to this change? 🤔

**Comment by jaimergp** [comment]
@goanpeca - not our fault. Both `conda` and `micromamba` fail to accept the offending file:

```
$ micromamba env create -f etc/example-empty-channels-environment.yml
info     libmamba ****************** Backtrace Start ******************
debug    libmamba Loading configuration
trace    libmamba Compute configurable 'create_base'
trace    libmamba Compute configurable 'no_env'
trace    libmamba Compute configurable 'no_rc'
trace    libmamba Compute configurable 'rc_files'
trace    libmamba Compute configurable 'root_prefix'
trace    libmamba Get RC files configuration from locations up to HomeDir
trace    libmamba Configuration not found at '/Users/jrodriguez/.mambarc'
trace    libmamba Configuration found at '/Users/jrodriguez/.condarc'
trace    libmamba Configuration not found at '/Users/jrodriguez/.conda/condarc.d'
trace    libmamba Configuration not found at '/Users/jrodriguez/.conda/condarc'
trace    libmamba Configuration not found at '/Users/jrodriguez/.conda/.condarc'
trace    libmamba Configuration not found at '/Users/jrodriguez/micromamba/.mambarc'
trace    libmamba Configuration not found at '/Users/jrodriguez/micromamba/condarc.d'
trace    libmamba Configuration not found at '/Users/jrodriguez/micromamba/condarc'
trace    libmamba Configuration not found at '/Users/jrodriguez/micromamba/.condarc'
trace    libmamba Configuration not found at '/var/lib/conda/.mambarc'
trace    libmamba Configuration not found at '/var/lib/conda/condarc.d/'
trace    libmamba Configuration not found at '/var/lib/conda/condarc'
trace    libmamba Configuration not found at '/var/lib/conda/.condarc'
trace    libmamba Configuration not found at '/etc/conda/.mambarc'
trace    libmamba Configuration not found at '/etc/conda/condarc.d/'
trace    libmamba Configuration not found at '/etc/conda/condarc'
trace    libmamba Configuration not found at '/etc/conda/.condarc'
trace    libmamba Update configurable 'no_env'
trace    libmamba Compute configurable 'file_specs'
error    libmamba Could not read 'channels' as vector of strings from '/Users/jrodriguez/devel/setup-miniconda/etc/example-empty-channels-environment.yml'
critical libmamba yaml-cpp: error at line 5, column 1: bad conversion
info     libmamba ****************** Backtrace End ********************
```
```
$ conda env create -f etc/example-empty-channels-environment.yml
Retrieving notices: ...working... done

SpecNotFound: /Users/jrodriguez/devel/setup-miniconda/etc/example-empty-channels-environment.yml is not a valid yaml file.
```

That test should be expected to fail. Maybe old versions supported that, but not anymore. So either a bug to report or something not supported anymore, but nothing we need to do on our side to handle it (other than xfail/remove the test).

**Comment by dbast** [comment]
libmamba is the default solver for conda since version 23.10.0 and miniconda3 23.10.0 ... so the new standard / reality is lib/mamba and users get used to it... do we still want this option to be able to switch to the old solver OR should we skip the option for the sake of less configuration hell and test-combination explosion?... users can also switch to the old solver by requesting an old miniconda3.

@jaimergp The PR looks good and is all green. Do you still want it? Merge or close it?

**Review by dbast — APPROVED** [review]

**Pull request merged**